### ISSUE TITLE
After 0.57.8 → 0.57.9, provider registry Locked

### ISSUE THREAD

**Issue by ignatenkobrain** [gruntwork-io/terragrunt#3100]
## Describe the bug

We are using v0.57.5 in our environment and now preparing upgrade to v0.57.9 however run-all plan is failing on some modules with following error:

```
time=2024-04-26T06:09:46Z level=info msg=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] ╷
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] │ Error: Failed to install provider
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] │ 
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] │ Error while installing snowflake-labs/snowflake v0.89.0: could not query
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] │ provider registry for registry.terraform.io/snowflake-labs/snowflake: 423
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] │ Locked
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] ╵
[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_dev/github-runner-admin] 
```

We are using terragrunt provider cache and with current version we've never seen this issue.

## Steps To Reproduce

I don't really have some minimalistic reproducer becuase on simple configuration it works fine on my laptop. This maybe has something to do with different lockfiles in different folders and how dependencies are used (e.g. I see this error on all modules using snowflake provider).

## Expected behavior

Should work as before.

## Nice to haves

- [x] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: 0.57.9
- OpenTofu/Terraform version: 1.8.2
- Environment details (Ubuntu 20.04, Windows 10, etc.):

**Comment by ignatenkobrain** [comment]
Update: 0.57.8 is working fine so something is broken in 0.578.9

**Comment by levkohimins** [comment]
Hi @ignatenkobrain,
We changed the approach to caching providers in version 0.57.9. 
Previously we used `terrafrom providers lock -platfrom=xxx_xxx` to start caching and then `terrafrom init`, now we use the same `terrafrom init` command twice but with different CLI configuration files. The new approach has many advantages, so rollback is not considered.

The error you mentioned
```
│ Error: Failed to install provider
│ 
│ Error while installing snowflake-labs/snowflake v0.89.0: could not query
│ provider registry for registry.terraform.io/snowflake-labs/snowflake: 423
│ Locked
```
is always returned the first time  `terraform init` is run, which means Terragrunt Provider Cache started caching. so it is ignored and hided from the logs. The second run we use different CLI config where Terragrunt Provider Cache acts as a proxy and does not return a such error.

I'll try to make a minor improvement in the log parsing and the way to pass of the CLI config file. But I can’t say for sure that this will help until I can reproduce this error.

**Comment by ignatenkobrain** [comment]
@levkohimins I don't mind some clutter in logs however what should I do to our configuration to avoid this error? I mean, the run-all plan here helps so should I fix / update lockfile or run some commands differently or? Sadly our configuration is not public so I can't share but I am happy to test any patches.

**Comment by ignatenkobrain** [comment]
@levkohimins Hope this helps to gather more info: 
[8_terragrunt run-all.txt](https://github.com/gruntwork-io/terragrunt/files/15133134/8_terragrunt.run-all.txt)

**Comment by ignatenkobrain** [comment]
This is happenning only for snowflake provider, all others are completely fine.... which brings me to a point... it uses Pascal-Case (or whatever it is called) in the URL (= `Snowflake-Labs`), can it be somehow problematic for terragrunt to cache it?

**Comment by ignatenkobrain** [comment]
I can't reproduce it locally but easily reproducible in our gh actions

**Comment by ignatenkobrain** [comment]
I've tried also with parallelism=1 and it still behaves like that.

<details>
<summary>Relevant logs</summary>

```
2024-04-27T05:53:24.3993991Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4000122Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 0): common_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4008994Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4017449Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 2): account_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4019885Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 2 locals (remaining 0): account_name, state_config prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4023675Z time=2024-04-27T05:53:24Z level=debug msg=[Partial] Included config /runner/_work/iac/iac/snowflake/cloud.hcl has strategy shallow merge: merging config in (shallow). prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4289655Z time=2024-04-27T05:53:24Z level=debug msg=terraform version: 1.8.2 prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4293494Z time=2024-04-27T05:53:24Z level=debug msg=Reading Terragrunt config file at /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/terragrunt.hcl prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4296590Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4317754Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 0): common_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4332597Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4341270Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 2): account_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4344472Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 2 locals (remaining 0): state_config, account_name prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4348119Z time=2024-04-27T05:53:24Z level=debug msg=[Partial] Included config /runner/_work/iac/iac/snowflake/cloud.hcl has strategy shallow merge: merging config in (shallow). prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4352090Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4365800Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 0): common_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4374505Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4406792Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 2): account_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4408593Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 2 locals (remaining 0): account_name, state_config prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4412807Z time=2024-04-27T05:53:24Z level=debug msg=Included config /runner/_work/iac/iac/snowflake/cloud.hcl has strategy shallow merge: merging config in (shallow) for dependency. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4425854Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4430552Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 2): account_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4433346Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 2 locals (remaining 0): account_name, state_config prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4439481Z time=2024-04-27T05:53:24Z level=debug msg=Found locals block: evaluating the expressions. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4444479Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 1 locals (remaining 2): account_vars prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4446308Z time=2024-04-27T05:53:24Z level=debug msg=Evaluated 2 locals (remaining 0): account_name, state_config prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4451851Z time=2024-04-27T05:53:24Z level=debug msg=Included config /runner/_work/iac/iac/snowflake/cloud.hcl has strategy shallow merge: merging config in (shallow). prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.4466983Z time=2024-04-27T05:53:24Z level=info msg=Downloading Terraform configurations from git::ssh://git@github.com/gooddata/terraform-snowflake-modules.git?ref=v0.2.4 into /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.8179643Z time=2024-04-27T05:53:24Z level=debug msg=Copying files from /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin into /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.8185933Z time=2024-04-27T05:53:24Z level=debug msg=Setting working directory to /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:53:24.8190028Z time=2024-04-27T05:53:24Z level=info msg=Caching terraform providers for /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin
2024-04-27T05:53:25.2160435Z time=2024-04-27T05:53:25Z level=debug msg=Acquiring lock file /home/runner/.cache/terragrunt/archives/registry.terraform.io-snowflake-labs-snowflake-0.89.0-linux_amd64.lock
2024-04-27T05:53:25.2162491Z time=2024-04-27T05:53:25Z level=debug msg=Fetching provider snowflake-labs/snowflake v0.89.0
2024-04-27T05:53:25.2205676Z time=2024-04-27T05:53:25Z level=error msg=Module /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin has finished with an error: 1 error occurred:
2024-04-27T05:53:25.2207607Z 	* [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin] exit status 1
2024-04-27T05:53:25.2208865Z 
2024-04-27T05:53:25.2209364Z  prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
```

```
2024-04-27T05:59:57.2760865Z time=2024-04-27T05:53:24Z level=debug msg=Generated file /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin/_provider.tf. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:59:57.2762370Z time=2024-04-27T05:53:24Z level=debug msg=Generated file /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin/_backend.tf. prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:59:57.2763187Z time=2024-04-27T05:53:24Z level=debug msg=Running command: /home/runner/.local/bin/terraform init prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:59:57.2763629Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m[31m╷[0m[0m
2024-04-27T05:59:57.2764290Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m│[0m [0m[1m[31mError: [0m[0m[1mFailed to install provider[0m
2024-04-27T05:59:57.2764696Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m│[0m [0m
2024-04-27T05:59:57.2765429Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m│[0m [0m[0mError while installing snowflake-labs/snowflake v0.89.0: could not query
2024-04-27T05:59:57.2766148Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m│[0m [0mprovider registry for registry.terraform.io/snowflake-labs/snowflake: 423
2024-04-27T05:59:57.2766581Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m│[0m [0mLocked
2024-04-27T05:59:57.2767032Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [31m╵[0m[0m
2024-04-27T05:59:57.2767409Z [/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] [0m[0m
2024-04-27T05:59:57.2768928Z time=2024-04-27T05:53:25Z level=error msg=terraform invocation failed in /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
2024-04-27T05:59:57.2770905Z time=2024-04-27T05:53:25Z level=debug msg=Copying lock file from /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin/.terragrunt-cache/5nYGFbkTH1ifyBpoT7oX6Dxsycw/joswRRlSlNjBCU0tFdPnMPkAZrU/github-runner-admin/.terraform.lock.hcl to /runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin prefix=[/runner/_work/iac/iac/snowflake/hqltkfn-aws_eu_west_1_prod/github-runner-admin] 
```

</summary>
</details>

**Comment by ignatenkobrain** [comment]
I am actually able to reproduce it locally too now. It is due to the `--terragrunt-include-module-prefix`

**Comment by ignatenkobrain** [comment]
[no-prefix.txt](https://github.com/gruntwork-io/terragrunt/files/15137246/no-prefix.txt)
[with-prefix.txt](https://github.com/gruntwork-io/terragrunt/files/15137247/with-prefix.txt)

**Comment by ignatenkobrain** [comment]
```diff
-               cmdStderr := io.MultiWriter(withPrefix(errWriter, prefix), &stderrBuf)
+               cmdStderr := io.MultiWriter(withPrefix(errWriter, ""), &stderrBuf)
```

in the `shell/run_shell_cmd.go` fixes the issue.

**Comment by ignatenkobrain** [comment]
https://github.com/gruntwork-io/terragrunt/pull/3105 fixes the issue.

**Comment by levkohimins** [comment]
Resolved in [v0.57.13](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.57.13) release.
